Caffe Yolo v2 training patch

SowmyaDhanapal opened the pull request:
@vera121 This patch will add support to prune Yolo v2 model using Synopsys-Caffe. It also contains support to generate LMDB to train Yolo v2 model.

SowmyaDhanapal committed:
Add Yolo-v2 training support

SowmyaDhanapal committed:
Fix XML parser and update function declaration

Generalize XML parsing for VOC2007 and VOC2012 datasets to avoid
updating difficult label in bounding box information without
reading from its corresponding XML file.

Update function declaration based on suggestion.

vera121 commented:
Hi @SowmyaDhanapal , please recheck your commit, the travis build result showed that there might be a bug:
In file included from ./include/caffe/common.hpp:22:0,
                 from ./include/caffe/blob.hpp:8,
                 from ./include/caffe/layers/yolo_v2_loss_layer.hpp:7,
                 from src/caffe/layers/yolo_v2_loss_layer.cpp:1:
src/caffe/layers/yolo_v2_loss_layer.cpp:374:5: error: no ‘void caffe::YoloV2LossLayer<Dtype>::Forward_gpu(const std::vector<caffe::Blob<Dtype>*>&, const std::vector<caffe::Blob<Dtype>*>&)’ member function declared in class ‘caffe::YoloV2LossLayer<Dtype>’
     STUB_GPU(YoloV2LossLayer);
     ^
src/caffe/layers/yolo_v2_loss_layer.cpp:374:5: error: no ‘void caffe::YoloV2LossLayer<Dtype>::Backward_gpu(const std::vector<caffe::Blob<Dtype>*>&, const std::vector<bool>&, const std::vector<caffe::Blob<Dtype>*>&)’ member function declared in class ‘caffe::YoloV2LossLayer<Dtype>’
     STUB_GPU(YoloV2LossLayer);
     ^
make: *** [.build_release/src/caffe/layers/yolo_v2_loss_layer.o] Error 1
make: *** Waiting for unfinished jobs....

fanghuaqi commented:
See travis job log: https://travis-ci.org/foss-for-synopsys-dwc-arc-processors/synopsys-caffe/jobs/415887020

SowmyaDhanapal committed:
Comment STUB_GPU(YoloV2LossLayer)

Since there is no GPU implementation for YoloV2LossLayer, commented
the STUB_GPU call to avoid build issues
